main.go

```diff
@@ -213,20 +228,32 @@ func diffIDs(sourceURL, destURL string) error {
 
 	}
 
-	for s, _ := range dests {
+	for s := range dests {
 		output.OnlyInDestination = append(output.OnlyInDestination, s)
 	}
 
 	return json.NewEncoder(os.Stdout).Encode(output)
 
 }
 
-func syncIDs(sourceURL, destURL string, deletes bool) error {
-	sourceIDs := make(chan string)
-	go fetchIDList(sourceURL, sourceIDs)
+type syncConfig struct {
```

Comment:
I think this is more than a simple config, having fields like `sourceIdsRetriever` & `destIdsRetriever`. Maybe a `syncService`?